### DISCUSSION TITLE
(Partent -> Child1 -> Child2) can not model this relationship with JpaContext

### DISCUSSION THREAD

**Discussion by asicode24** [mapstruct/mapstruct#2978] [General]
With JpaContext Class logic, We are not able to model the Parent->child1->child2 relation. Mapstruct generated implemented class logic works only if both child1 and child2 are the children of the Parent

Here is the link of the example using JPA Context.

https://github.com/mapstruct/mapstruct-examples/blob/main/mapstruct-jpa-child-parent/src/main/java/org/mapstruct/jpa/JpaContext.java

**Comment by filiphr** [comment]
The `JpaContext` class you are referring to is in the MapStruct examples repository. It is meant to be used as an example and as an inspiration for how to achieve certain things. If you are looking for something else, then I would suggest implementing it in a way that achieves your purpose.

e.g. perhaps you'll need to have a `setEntity` with `child1` as well and establish relation for child1 and child2 and establish the relation appropriately

**Comment by fajksk-dev** [comment]
Is there any way to define parent-child relation mapping together with spring conversion service integration. I was trying to do it, but so far without any success, because interfaces needs to extend Converter interface from spring where there is one argument method convert and in order to achieve effect from provided example somehow we need to pass @Context.
